internal/testutil/test_hcls.go

```diff
@@ -0,0 +1,142 @@
+// ###### RESOURCES CONSTs
+// ########################################
+
+const pfPort = "packetfabric_port"
+const pfCloudRouter = "packetfabric_cloud_router"
+const pfCloudRouterConnAws = "packetfabric_cloud_router_connection_aws"
+const pfCloudRouterBgpSession = "packetfabric_cloud_router_bgp_session"
+const pfCsAwsHostedConn = "packetfabric_cs_aws_hosted_connection"
+
+// ########################################
+// ###### HCLs FOR REQUIRED FIELDS
+// ########################################
+
+// packetfabric_port
+func HclPort() (hcl, resourceName, resourceReferece string) {
+	resourceReferece, resourceName = _generateResourceName(pfPort)
+	pop, _, media, availableSpeed, err := GetPopAndZoneWithAvailablePort(os.Getenv(PF_CRC_SPEED_KEY))
+	if err != nil {
+		log.Panic(err)
+	}
+	var portEnabled bool
+	if os.Getenv(PF_PORT_ENABLED_KEY) != "" {
+		portEnabled, err = strconv.ParseBool(os.Getenv(PF_PORT_ENABLED_KEY))
+		if err != nil {
+			log.Fatal("port enabled must be either true or false")
+		}
+	}
+	hcl = fmt.Sprintf(
+		RResourcePort,
+		resourceName,
+		_generateUniqueNameOrDesc(pfPort),
+		media,
+		pop,
+		availableSpeed,
+		os.Getenv(PF_PORT_SUBTERM_KEY),
+		portEnabled)
+	return
+}
+
+// packetfabric_cloud_router
+func HclCloudRouter() (hcl string, resourceName string) {
+	resourceName, hclName := _generateResourceName(pfCloudRouter)
+	hcl = fmt.Sprintf(
+		RResourcePacketfabricCloudRouter,
+		hclName,
+		_generateUniqueNameOrDesc(pfCloudRouter),
+		os.Getenv(PF_ACCOUNT_ID_KEY),
+		os.Getenv(PF_CR_CAPACITY_KEY),
+		os.Getenv(PF_CR_MEMBER_KEY))
+	return
+}
+
+// packetfabric_cloud_router_connection_aws
+func HclCloudRouterConnectionAws() (hcl string, crResourceName, resourceName string) {
+
+	hclCloudRouter, crResourceName := HclCloudRouter()
+	resourceName, hclName := _generateResourceName(pfCloudRouterConnAws)
+	pop, _, _, _, err := GetPopAndZoneWithAvailablePort(os.Getenv(PF_CRC_SPEED_KEY))
+	if err != nil {
+		log.Fatal(err)
+	}
+	crcHcl := fmt.Sprintf(
+		RResourceCloudRouterConnectionAws,
+		hclName,
+		crResourceName,
+		os.Getenv(PF_CRC_AWS_ACCOUNT_ID_KEY),
+		os.Getenv(PF_ACCOUNT_ID_KEY),
+		_generateUniqueNameOrDesc(pfCloudRouterConnAws),
+		pop,
+		os.Getenv(PF_CRC_SPEED_KEY))
+
+	hcl = fmt.Sprintf("%s\n%s", hclCloudRouter, crcHcl)
+	return
+}
+
+// packetfabric_cloud_router_bgp_session
+func HclBgpSession() (hcl string, resourceName string) {
+
+	// hclCR, crResourceName := HclCloudRouter()
+	hclRCConn, crResourceName, crConnResourceName := HclCloudRouterConnectionAws()
+
+	resourceName, hclName := _generateResourceName(pfCloudRouterBgpSession)
+	bgpSessionHcl := fmt.Sprintf(
+		RResourceCloudRouterBgpSession,
+		hclName,
+		crResourceName,
+		crConnResourceName,
+		os.Getenv(PF_CRBS_ADDRESS_FMLY_KEY),
+		os.Getenv(PF_CRBS_MHTTL_KEY),
+		os.Getenv(PF_CRBS_REMOTE_ASN_KEY),
+		os.Getenv(PF_CRBS_PRFX1_KEY),
+		os.Getenv(PF_CRBS_TYPE1_KEY),
+		os.Getenv(PF_CRBS_PRFX2_KEY),
+		os.Getenv(PF_CRBS_TYPE2_KEY))
```

Comment:
Why should we provide BGP session data via env and not in the test code? I see no reason to test different BGP configs via Terraform tests. They can be hardcoded. Overparameterization is also evil 😈

Reply:
I think our plan was to remove all hardcoded values and avoid it as much as possible. It's also easier change values if needed in the future without code interventions.

Reply:
Sounds good then! So, please share the hardcoded values as soon as possible, so we can update this PR accordingly.